File: tests/uasm/sample/early_return_guard/EarlyReturnGuardTest.ts
import { UdonBehaviour } from "@ootr/udon-assembly-ts/stubs/UdonDecorators";
import { UdonSharpBehaviour } from "@ootr/udon-assembly-ts/stubs/UdonSharpBehaviour";
import { Debug } from "@ootr/udon-assembly-ts/stubs/UnityTypes";

@UdonBehaviour()
export class EarlyReturnGuardTest extends UdonSharpBehaviour {
  private isReady = false;

  Start(): void {
    if (!this.isReady) {
      Debug.Log("skip");
      return;
    }

    Debug.Log("run");
  }
}

File: tests/uasm/sample/early_return_guard/EarlyReturnGuardTest.cs
using UdonSharp;
using UnityEngine;

[UdonBehaviourSyncMode(BehaviourSyncMode.None)]
public class EarlyReturnGuardTest : UdonSharpBehaviour
{
    private bool isReady = false;

    public void Start()
    {
        if (!isReady)
        {
            Debug.Log("skip");
            return;
        }

        Debug.Log("run");
    }
}
